### DISCUSSION TITLE
Code signing certificates - Certificate type

### DISCUSSION THREAD

**Discussion by burekas7** [codemagic-ci-cd/codemagic-docs#2177] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Code signing certificates - Certificate type

1. Is there a difference if the Certificate type is "Development" or "Production"?
Because I don't succeed to create a .p12 file which is "Development" type.
Every trying of another certificate type in apple and upload it to CodeMagic shows "Production" type.
 
2. How can I generate a certificate that will show it as "Development" type?

3. When generating a Certificate what to choose: 
Apple Development / iOS App Development / Apple Distribution / iOS Distribution (App Store and Ad Hoc) ?

4. Do I need to attach both Distribution and Development certificates, 
or is it enough to add only one of them no matter what type is?

**Which mobile framework are you using?**

Flutter (Dart)

**Comment by dtrdic** [comment]
hey @burekas7, the process of creating a .p12 file (or exporting a certificate) is essentially the same for both "Development" and "Production" certificates. Checkout [this link](https://support.magplus.com/hc/en-us/articles/203808748-iOS-Creating-a-Distribution-Certificate-and-p12-File)
When you create a .p12 file for a "Development" certificate, it's primarily for testing on devices registered with your developer account. You would use the "Production" certificate when you're ready to submit your app to the App Store or distribute it to end-users.

Check also this [step-by-step guide](https://labs.codemagic.io/your-first-flutter-app-to-appstore/#3) on how to publish your app to App Store, from start to finish

**Reply by burekas7** [reply]
@dtrdic 

Thanks for your reply.

I know how to create a Certificate and .p12 file.

The reason I asked is because by now the current attached certificated in CodeMagic was of "Developer" type, and basically I never use CodeMagic to upload my app directly to the stores, I upload it first to the Testing sections (TestFlight, Google Play internal).
1) For this case do I need a "Development" certificate or "Production"?
2) Can I use "Production" type also for my case?

Also I found that in order to create a "Development" certificate, it can be done only by using the Xcode.
Because every certificate that has been done by AppstoreConnect was a type of "Production" when I add it to CodeMagic.

So my question is which kind of Cetificate do I actually need here?
Apple Development / iOS App Development / Apple Distribution / iOS Distribution (App Store and Ad Hoc)

Thanks.

**Reply by dtrdic** [reply]
use this one; iOS Distribution (App Store and Ad Hoc)

**Reply by burekas7** [reply]
@dtrdic 
Thanks.

And what about the first part questions?

**Reply by dtrdic** [reply]
Hey @burekas7, yes you can use this Production type certificate for testing as well, this is how I'm using it also

**Comment by dtrdic** [comment]
I'm tentatively closing the discussion due to a lack of activity. You can open the discussion again by replying to this message or opening a new discussion.
